wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -336,6 +339,35 @@ deriveJSON Serokell.defaultOptions ''AccountUpdate
 instance Arbitrary AccountUpdate where
   arbitrary = AccountUpdate . fromString <$> pure "myAccount"
 
+-- | Summary about single address.
+data AddressInfo = AddressInfo
+  { addrId       :: !Core.Address
+  , addrBalance  :: !Core.Coin
+  , addrIsUsed   :: !Bool
+  , addrIsChange :: !Bool
+  } deriving (Show, Generic)
```

Comment:
Naming is hard, and I'm fairly happy for now with the notion of an `AddressInfo`, but just bikeshedding on how to call this type: `WalletAddress` maybe? It might convey a little bit better the fact this is conceptually the wallet representation of an `Address` not a `Core.Address` per se.

I would also dropped the `Is` from `IsUsed`. Sending as JSON something like `{"used": true}` is perfectly self-explanatory. I would also rename `addrIsChange` to `addrIsChangeAddress`, is a bit more self-explanatory. In a nutshell, I might dare to suggest to go with:

```haskell
data WalletAddress = WalletAddress
  { addrId       :: !Core.Address
  , addrBalance  :: !Core.Coin
  , addrUsed   :: !Bool
  , addrIsChangeAddress :: !Bool
 } deriving (Show, Generic)
```

This would generate something like:

```
{ "id": "AABBCC"
, "balance": 1000
, "used": true
, "isChangeAddress": false
}
```

What do you think?

Reply:
Yeah, this way it looks much bette, thanks for the note!